### PR TITLE
Delete items from group

### PR DESCRIPTION
**Pull request by lepmik**
excuse the bad commit messages, will squash when merge.

### CODE REVIEW

**Comment by lepmik** [comment]
and... excuse the `__str__` implementation on dataset, it does not belong in this PR, but I think it is fine to leave it here, agree?

**Comment by lepmik** [comment]
#67

File: tests/test_group.py [hunk 1]
```diff
@@ -176,37 +176,86 @@ def test_set_item_intermediate(exdir_tmpfile):
...
+    del grp["foo"]
+    assert "foo" not in grp
+
+
+def test_nonexisting(setup_teardown_file):
+    """Deleting non-existent object raises KeyError."""
+    f = setup_teardown_file[3]
+    grp = f.create_group("test")
+
+    with pytest.raises(KeyError):
```
Comment (miladh) [inline tests/test_group.py]: Consider using `match` arg in `raises` to also check the error message

**Review by miladh — APPROVED** [review]

File: tests/test_group.py (hunk 1 above)
Comment (lepmik) [reply tests/test_group.py]: good idea!

**Pull request merged**